changelog/13047.bugfix.rst

```diff
@@ -0,0 +1 @@
+* Fixed an issue in equality checks where `bool` and `numpy.bool_` types were not handled consistently. This ensures proper handling of native and numpy boolean types. See #13047.
```

Comment:
The second sentence seems a bit redundant to me. Also no need to mention the issue number, that is added automatically based on the filename. :+1: 

```suggestion
* Fixed an issue in equality checks where `bool` and `numpy.bool_` types were not handled consistently. 
```